grammar.js

```diff
@@ -427,7 +430,7 @@ module.exports = grammar({
         _assignment_pattern: $ => seq(
             field("name", $._variable_name),
             PUNC().eq,
-            field("value", $.pipeline),
+            field("value", $._expression),
```

Comment:
starting from [#9589](https://github.com/nushell/nushell/pull/9589) and [#9658](https://github.com/nushell/nushell/pull/9658) 

```nushell
let x = ls | get name
# and
mut x = ls | get name
```

are legal

```nushell
const x = some const | pipeline
```

is not ([yet?](https://github.com/nushell/nushell/pull/9499)) legal nushell, but i think the parser should be robust enough to handle it
```suggestion
            field("value", $._pipeline),
```

Reply:
No, these let, mut, const assignment with pipeline are not legal nu statement yet. At least nu 0.82.0 on my machine does not accept those as valid statements.

Reply:
But, yeah, maybe we can make our parser support that.

Reply:
we generally use the latest nushell main to add changes. if a syntax change is pushed to main we implement it way before the release. that way when a release is out, the parser already supports any new syntax

this change is recent, but i saw it coming :) that's why the parser already handles it. table annotations are also recent and have been already [implemented](https://github.com/nushell/tree-sitter-nu/pull/20)

i think by being one step ahead of the release, we can reduce surprises to people who use the parser for syntax highlighting

Reply:
Done, I've change `_expression` back to `pipeline`, tests also get reverted to previous version.